# Week 6 Fantasy Recap

Alright gents, another week in the book. For those of you unaware, you are reading the prose of our first '18-'19 season's two-time poster. Let me tell you; it is a miraculous and magical feeling. I was a bit concerned the past few weeks when my team decided to collectively shit the bed but with concerns regarding my fantasy performance repealing, let's get a data-driven look at Week 6 in fantasy land.

Gonna take a different, albeit equitable, perspective of last week's pigskin matches. Clearly, the name of the game in fantasy is to assemble a squad of heads that put up numbers week to week. But realistically, a single player's performance can depend *so heavily* on the other players on the field. And for that reason, I'm going to look at the total offensive production of the NFL from Week 6. Dive into the numbers. Do the `data` thing. Ja feel?

**Sidebar:** *I thought I'd be able to do a lot more with the data. But the teams names didn't align. I don't have the specific stats for each player from Yahoo. In all fairness anyone that posted after Butson's masterpiece was going to fall short, so might as well add to that diasappointment.*

Quick plug for my bo**Y** Buston. Knocked it out of the park last week with the freestyle post. Refined disses, production-level quality and dutiful attention to detail are things we can all get behind. Collectively as a fantasy league, this season marks a graduation or emergence of sorts when looking at the coveted weekly post.

In [7]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import matplotlib.dates as mdates
import matplotlib.ticker as mtick
from Matchup import parse_matchup, process_matchups
from matplotlib import cm
from datetime import datetime
import os
import urllib.request
%matplotlib inline
sns.set(style="ticks", color_codes=True)
pd.set_option('display.float_format', lambda x: '%.3f' % x)

In [8]:
qb = pd.read_excel(r'C:\Users\sdemarco\Desktop\wk6-stats.xlsx', header=0, sheet_name=0)
wr = pd.read_excel(r'C:\Users\sdemarco\Desktop\wk6-stats.xlsx', header=0, sheet_name=1)
rb = pd.read_excel(r'C:\Users\sdemarco\Desktop\wk6-stats.xlsx', header=0, sheet_name=2)
qb = qb[['Name', 'Date', 'Tm', 'Opp', 'Result', 'Cmp', 'Att', 'Cmp%', 'Yds', 'TD', 'Int', 'Rate', 'Sk']]
wr = wr[['Name', 'Date', 'Tm', 'Opp', 'Result', 'Tgt', 'Rec', 'Yds', 'Y/R', 'TD', 'Ctch%', 'Y/Tgt']]
rb = rb[['Name', 'Date', 'Tm', 'Opp', 'Result', 'Att', 'Yds', 'Y/A', 'TD']]

In [9]:
df = process_matchups('week_06')
df['dif'] = df.act - df.proj
df['pct_dif'] = df.dif / df.proj * 100

@barstoolbass VS We are the smes! complete.
Papa Kommy the Wise VS Pasta La Vista complete.
Scoopin' Dubs VS Big Data Little Dick complete.
The Old Bol & Chain VS Donny Moscow complete.
Time to Refi VS The Red Rockets complete.


# How this is gonna work

Alright, I scooped some data from the `! WORLD WIDE WEB :D !` and plan to go through offensive production in the NFL from three of the largest point-generating perspectives:
- Passing
- Rushing
- Receiving

Hopefully out of this we can start to understand that offensive production is something that each and every one of us is fighting for. In the words of our largest melanin-producing fantasy owner, "The bean is out there, you just gotta go get it." In other words, the combined production all NFL teams represents the maximum number of points one can expect to benefit from, from a fantasy perspective. Assemble a squad that takes the largest percentage of offensive production from each of the most productive offenses in the NFL and you've got a championship game on your horizon.

# Passing

From a passing perspective, we're gonna focus on yards and touchdowns. In Week 6, there were 7,805 total passing yards and 53 touchdown throws. This corresponds to *roughly* 312 fantasy points from the yardage and 212 from the TEET bois.

In [10]:
qb[['Cmp', 'Yds', 'TD']].sum()

Cmp     658
Yds    7805
TD       53
dtype: int64

In [26]:
tm = qb.groupby('Tm')['Yds', 'TD'].sum().reset_index()
tm.sort_values(by='Yds', ascending=False)[:3]

Tm  Yds  TD
10  GNB  425   2
27  TAM  395   4
17  MIA  380   3

In [24]:
tm.sort_values(by='TD', ascending=False)[:5]

Tm  Yds  TD
12  IND  301   4
27  TAM  395   4
14  KAN  352   4
25  SEA  222   3
5   CHI  316   3

### Bottom-line

Basically, the Packers, Bucs and Dolphins made up a lot of the 'ball-through-air' production this week, while the Colts, Bucs, 49ers, Seahawks and Bears had 3 or more TEET bois.

## Rushing

From a rushing perspective, we're gonna focus on yards and touchdowns. In Week 6, there were 3,743 yards on the ground and 24 rushing touchdowns. That makes up *roughly* 374.3 fantasy points from the ground on yardage and 144 from TEET bois.

In [12]:
rb[['Yds', 'TD']].sum()

Yds    3743
TD       24
dtype: int64

In [28]:
rush = rb.groupby('Tm')['Yds', 'TD'].sum().reset_index()
rush.sort_values(by='Yds', ascending=False)[:3]

Tm  Yds  TD
16  LAR  270   2
15  LAC  246   3
8   DAL  206   2

In [30]:
rush.sort_values(by='TD', ascending=False)[:5]

Tm  Yds  TD
15  LAC  246   3
19  NWE  173   3
2   BAL  123   2
24  PIT  112   2
8   DAL  206   2

### Bottom-line

THe Rams, Chargers and Cowgirls had most of their production on the ground, all sporting 200+ rushing yards. Rushing touchdowns included 2+ from the Chargers, Patriots, Ravens, Steelers and Cowgirls.

## Receiving

From a receiving perspective, we want to look at catches, yards and TEET bois. In Week 6, there were 658 receptions, representing that same 7,805 yards through the air. Of those receptions, 53 were TEET bois.

In [13]:
wr[['Rec', 'Yds', 'TD']].sum()

Rec     658
Yds    7805
TD       53
dtype: int64

In [38]:
catch = wr.groupby('Tm')['Rec', 'Yds', 'TD'].sum().reset_index()
catch.sort_values(by='Rec', ascending=False)[:3]

Tm  Rec  Yds  TD
24  PIT   32  369   1
1   ATL   31  354   3
27  TAM   30  395   4

In [39]:
catch.sort_values(by='Yds', ascending=False)[:3]

Tm  Rec  Yds  TD
10  GNB   25  425   2
27  TAM   30  395   4
17  MIA   28  380   3

In [40]:
catch.sort_values(by='TD', ascending=False)[:3]

Tm  Rec  Yds  TD
12  IND   23  301   4
27  TAM   30  395   4
14  KAN   23  352   4

### Bottom-line

The Steelers, Falcons and Bucs had the highest number of total receptions (30+). The Packers, Bucs and Dolphins had 380+ yards through the air. The Colts, Bucs, and 49ers all had 4 TEET bois in the air.

## WOW Steve! That was literally pointless!

Well, yeah. I know. What would've made this interesting is to include the proportion of offensive production each player is responsible for and seeing how those numbers relate to overall fantasy production. By drafting the heads who have the highest relative proportion of offensive production, do you guarantee more fantasy points? By playing two RB's from a squad that splits workloads between two backs and runs the ball (effectively) almost every play more beneficial than playing a single RB from a trash team that has a shitty O-line?

I was hoping to join the data from individual fantasy teams with the data from NFL teams but it didn't happen. There's no data out there.

This is the kinda garbage I was after. But maybe if I can lock down some better Foo'in data I'll be able to do that the next time I *inevitably* post again.